plugins/org.python.pydev.core/src/org/python/pydev/core/cache/DiskCache.java

```diff
@@ -119,9 +123,13 @@ public static DiskCache loadFrom(FastBufferedReader reader, ObjectsPoolMap objec
                                 }
                                 break;
                             case 3:
-                                //path in zip
-                                key.key = new ModulesKeyForZip(key.key.name, key.key.file,
-                                        ObjectsInternPool.internLocal(objectsPoolMap, buf.toString()), true);
+                                //regular folder path or zip path
+                                if (buf.toString().equals("|^")) {
```

Comment:
Instead of using `buf.toString().equals(...)`, I think you should use buf.endsWith(...).